### PR TITLE
FIX: Properly render HEAD with `NeogitBranchHead` highlight group

### PR DESCRIPTION
**Pull request by gollth**
Sorry I just noticed, I introduced a small bug in my previous PR #934, where local checked out branches are not rendered with the `NeogitBranchHead`. With a user config of:

```lua
local blue = "#bae7ff"
local remote = "#b4eeb2"

vim.api.nvim_set_hl(0, "NeogitBranch", { fg = blue, bold = true })
vim.api.nvim_set_hl(0, "NeogitBranchHead", { fg = blue, bold = true, underline = true })
vim.api.nvim_set_hl(0, "NeogitRemote", { fg = remote, italic = true })
```

and a branch called `foo` checked out, this is what happens:

|Before|After|
|:--|:--|
|before|after|

This PR should fix this now =)

### CODE REVIEW

File: lua/neogit/buffers/common.lua [hunk 1]
```diff
@@ -113,13 +113,13 @@ M.CommitEntry = Component.new(function(commit, args)
 
   -- Parse out ref names
   if args.decorate and commit.ref_name ~= "" then
-    local ref_name, _ = commit.ref_name:gsub("HEAD %-> ", "")
-    local is_head = string.match(commit.ref_name, "HEAD %->") ~= nil
-    local branch_highlight = is_head and "NeogitBranchHead" or "NeogitBranch"
-
-    local info = git.log.branch_info(ref_name, git.remote.list())
+    local info = git.log.branch_info(commit.ref_name, git.remote.list())
+    -- print(">> " .. vim.inspect(commit.ref_name) .. " (" .. vim.inspect(is_head) .. "): " .. vim.inspect(info))
```
Comment (CKolkey) [inline lua/neogit/buffers/common.lua:117]: Probably can remove this?

**Review by CKolkey — CHANGES_REQUESTED** [review]
Looks great, lets just remove the commented out line and we're good :)

**Comment by gollth** [comment]
Dammit I noticed another bug, while retesting this... 

Right now there is now way to display _remote only_ branches. My impl will always highlight the last part of each branch with the `NeogitBranch` highlight group:

| Before | After |
|:--|:--|
|before|after|

Note in the example above, how `github/master` and `gollth/master` are both highlighted in green/blue indicating that my local `master` branch is both at `github/master` & `gollth/master` AT THE SAME TIME !?! Same goes with with the `fix-head-log-graph` being locally a commit further than its remote AND SIMULTANEOUSLY at its remote...

This should be fixed now as well, sorry for all the confusion @CKolkey. Git is hard... 🙈

**Comment by CKolkey** [comment]
Haha, no joke. The last 10~ months of working on this project have been a great deep-dive into _so much_ of how git works under the hood, and I still feel like I know almost nothing. So many small details.

**Comment by gollth** [comment]
I can imagine o.O This Torvalds dude is quite the smart guy

**Comment by CKolkey** [comment]
No joke. Once (linux) is good luck. Twice (git)... impressive. I watched a talk he gave at google in like... 2010 maybe, the very early days of git. Very interesting to see him explain the shortcomings of every other VCS available, things that I definitely take for granted like "easy branching" and "fully decentralised". Good stuff.

**Pull request merged**